engine/src/main/java/org/camunda/bpm/engine/impl/cmmn/deployer/CmmnDeployer.java

```diff
@@ -70,6 +92,69 @@ protected void addDefinitionToDeploymentCache(DeploymentCache deploymentCache, C
     deploymentCache.addCaseDefinition(definition);
   }
 
+  protected JobDefinitionManager getJobDefinitionManager() {
+    return getCommandContext().getJobDefinitionManager();
+  }
+
+  @Override
+  protected void definitionAddedToDeploymentCache(DeploymentEntity deployment, CaseDefinitionEntity definition, Properties properties) {
+    List<JobDeclaration<?, ?>> declarations = properties.get(JOB_DECLARATIONS_PROPERTY).get(definition.getKey());
+    if(declarations!=null && !declarations.isEmpty())updateJobDeclarations(declarations, definition, deployment.isNew());
+  }
+
+  protected void updateJobDeclarations(List<JobDeclaration<?, ?>> jobDeclarations, CaseDefinitionEntity definition, boolean isNewDeployment) {
+    if(jobDeclarations == null || jobDeclarations.isEmpty()) {
+      return;
+    }
+
+   final JobDefinitionManager jobDefinitionManager = getJobDefinitionManager();
+
+    if(isNewDeployment) {
+      for(JobDeclaration<?,?> jobDeclaration:jobDeclarations){
+        if(jobDeclaration instanceof TimerJobDeclaration){
+          createJobDefinition(definition, (TimerJobDeclaration<?>) jobDeclaration);
+        }else{
+          //TODO unable to handle this type of jobs, log warnings?
+          continue;
+        }
+      }
+      // create new job definitions:
+    } else {
+        //TODO more to be done here.
+      List<JobDefinitionEntity> existingDefinitions = jobDefinitionManager.findByCaseDefinitionId(definition.getId());
+      // TODO is this call required for CMMN? LegacyBehavior.migrateMultiInstanceJobDefinitions(processDefinition, existingDefinitions);
+      for (JobDeclaration<?, ?> jobDeclaration : jobDeclarations) {
+        boolean jobDefinitionExists = false;
+        for (JobDefinition jobDefinitionEntity : existingDefinitions) {
+
+          // <!> Assumption: there can be only one job definition per activity and type
+          if(jobDeclaration.getActivityId().equals(jobDefinitionEntity.getActivityId()) &&
+                  jobDeclaration.getJobHandlerType().equals(jobDefinitionEntity.getJobType())) {
+            jobDeclaration.setJobDefinitionId(jobDefinitionEntity.getId());
+            jobDefinitionExists = true;
+            break;
+          }
+        }
+        if(!jobDefinitionExists) {
+          // not found: create new definition
+          createJobDefinition(definition, (TimerJobDeclaration<?>) jobDeclaration);
+        }
+
+      }
+
+    }
+  }
+
+  private void createJobDefinition(CaseDefinitionEntity definition, TimerJobDeclaration<?> jobDeclaration) {
```

Comment:
Change the method signature to

```
protected void createJobDefinition(CaseDefinitionEntity definition, JobDeclaration<?, ?> jobDeclaration)
```

By the way instead of `private` we use `protected`. This give the users the possibility to override such methods to achieve custom behavior.